starlette/routing.py

```diff
@@ -184,15 +184,18 @@ def matches(self, scope: Scope) -> typing.Tuple[Match, Scope]:
                     return Match.FULL, child_scope
         return Match.NONE, {}
 
-    def url_path_for(self, name: str, **path_params: str) -> URLPath:
-        seen_params = set(path_params.keys())
+    def url_path_for(self, *args: str, **kwargs: str) -> URLPath:
+        assert (
+            len(args) == 1
+        ), "Exactly one positional argument required: the routes name."
```

Comment:
How about we use...

```python
assert len(args) > 0, "Missing route name as the first argument."
assert len(args) < 2, "Invalid positional argument passed."
```

That'd be a bit more verbose, codewise, but it'd be nice since it's a user-facing error that's reasonably likely to be seen, and it'd be useful if we can be as helpful as possible.